### ISSUE TITLE
Crazy weights from sim.get_array_metadata()

### ISSUE THREAD

**Issue by Vinc0110** [NanoComp/meep#1469]
For the spatial integration of field components, Meep provides the spatial dimensions and the weights for the trapezoidal approximation in `sim.get_array_metadata()`. The user can then just get the integral from a weighted sum of the field components, according to [the documentation](https://meep.readthedocs.io/en/latest/Python_User_Interface/#array-metadata).

I've noticed strange results from those integrals in certain simulations. In the example script below, this is reproduced with a simple straight waveguide in 3D and a 1D DFT monitor for Ez across the waveguide. The length of that monitor in z is adjusted with `sz_dft`.

I'm running the serial pymeep in version `1.16.2-alpha.6814fc`, which is what I got when I updated today to the latest nightly build. There seems to be a mismatch of the version numbers (latest version on GitHub is supposed to be 1.17.1). 

For `resolution=5` and `sz_dft=3`, the weights and the integration results are clearly incorrect:
![int_ez_res5_3](https://user-images.githubusercontent.com/63644854/104129579-ba2cbe80-536c-11eb-861b-c8a3d18c9111.png)

Output of the weights:
`[1.46924108e+195 7.50198479e+247 2.00000000e-001 9.08367206e+223
 1.42709634e+248 4.00086553e+252 9.00118393e+223 1.77690664e+248
 9.00089641e+223 1.15314036e+214 3.17095868e+180 6.96742157e+252
 1.06314416e+248             nan 2.00000000e-001 2.35624291e+251]`

Moreover, the weights are **different for every run**, but the results from the integration with `trapz()` look fine every time.

When I change the size of the DFT monitor to `sz_dft=2`, things look better. Still, I see a difference between the results from the weighted sum and a direct integration using [NumPy's trapz() function](https://numpy.org/doc/stable/reference/generated/numpy.trapz.html):
![int_ez_res5_2](https://user-images.githubusercontent.com/63644854/104129523-95d0e200-536c-11eb-86d7-f66b6538ca13.png)

Output of the weights:
`[0.025 0.175 0.2   0.2   0.2   0.2   0.2   0.2   0.2   0.2   0.175 0.025]`

Interestingly, `resolution=6` and `sz_dft=1` has a good start and only turns bad in the last quarter:
![int_ez_res6_1](https://user-images.githubusercontent.com/63644854/104129780-00365200-536e-11eb-9d71-ec12ea3abaeb.png)

Output of the weights:
`[2.08333333e-02 1.45833333e-01 1.66666667e-01 1.66666667e-01
 1.66666667e-01            nan 6.15366081e+98 1.16024984e+02]`

Example script:

```
import meep as mp
import numpy as np
import matplotlib.pyplot as mplt

resolution = 6
f_center = 0.2
f_width = 0.1
n_freqs = 100
sz_dft = 1

geometry = [mp.Block(mp.Vector3(mp.inf, 2, 1), center=mp.Vector3(), material=mp.Medium(epsilon=12))]
sources = [mp.Source(mp.GaussianSource(frequency=f_center, fwidth=f_width),
                     component=mp.Ez,
                     center=mp.Vector3(-6, 0, 0),
                     size=mp.Vector3(0, 3, 1))]
cell = mp.Vector3(16, 8, 5)
sim = mp.Simulation(cell_size=cell,
                    boundary_layers=[mp.PML(1.0)],
                    geometry=geometry,
                    sources=sources,
                    resolution=resolution)
ez_monitor = sim.add_dft_fields([mp.Ez], f_center, f_width, n_freqs,
                                center=mp.Vector3(6, 0, 0), size=mp.Vector3(0, 0, sz_dft))
sim.run(until_after_sources=100)

(x, y, z, w) = sim.get_array_metadata(dft_cell=ez_monitor)
print()
print(w)
print()
print(z)
i_freq = int(n_freqs / 2)
ez_integral_sum = np.zeros(len(z), dtype=np.complex)
ez_integral_trapz = np.zeros(len(z), dtype=np.complex)
ez = sim.get_dft_array(ez_monitor, mp.Ez, i_freq)
for i, z_i in enumerate(z):
    ez_integral_sum[i] = np.sum(w[:i] * ez[:i])
    ez_integral_trapz[i] = np.trapz(ez[:i], z[:i])

mplt.figure()
mplt.title('sz = {}, resolution = {}'.format(sz_dft, resolution))
mplt.plot(z, np.abs(ez_integral_sum), label='sum()')
mplt.plot(z, np.abs(ez_integral_trapz), label='trapz()')
mplt.xlabel('z')
mplt.ylabel('$\int E_\mathrm{z}(z) \, \mathrm{d}z$')
mplt.legend(loc='best')
mplt.tight_layout()
mplt.show()
```

**Comment by oskooi** [comment]
There was a bug in `get_array_metadata` related to empty dimensions (as in your example involving a 1d slice in a 3d cell) which was fixed in #1456. This was released with [version 1.17.0](https://github.com/NanoComp/meep/releases/tag/v1.17.0). Unfortunately, the [nightly built Conda package](https://anaconda.org/simpetus/pymeep) is still version 1.16.2 and has not yet been updated. The [Conda package for the official release](https://anaconda.org/conda-forge/pymeep) has been updated to 1.17.0 but only for Linux and not macOS. In any event, you will need to use at least version 1.17.0 for the bug fix.

**Comment by Vinc0110** [comment]
Alright, thanks! I can confirm that the bug is fixed in version 1.17 (I'm using Linux).
It is weird, though, that the nightly build is not up to date, while the official release is...
